### PR TITLE
Update

### PR DESCRIPTION
**Pull request by axelchalon**

### CODE REVIEW

File: scripts/generate-config-data.js [hunk 1]
```diff
@@ -63,9 +66,16 @@ function parseDefaultValue (defaultValue) {
       return [];
     default:
       try {
+        // Remove thousands separators if numeric value
+        if (defaultValue.match(/^[\d_]+$/)) {
```
Comment (dvdplm) [inline scripts/generate-config-data.js]: This regexp matches on wacky strings like e.g. `________000__` – not sure if that's a big deal, but you might want to lock it down a little? Also, perhaps `defaultValue.test(……)` would work here? Or even skip the test entirely and just `replace()`?

File: scripts/generate-config-data.js [hunk 2]
```diff
@@ -117,7 +127,7 @@ function makeCliConfigTree (parsedCliOptions) {
 function getStructFields (name, source) {
   const subsource = source.match(new RegExp(`^struct ${name} {[^]+?}`, 'm'))[0];
 
-  const fields = subsource.split('\n').slice(1, -1);
+  const fields = subsource.split('\n').slice(1, -1).filter(s => s.trim() && !s.trim().startsWith('#') && !s.trim().startsWith('_legacy'));
```
Comment (dvdplm) [inline scripts/generate-config-data.js:130]: The first `trim()` looks superfluous. How about: `s => s=s.trim() && !s.startsWith(…) && !s.startsWith(…)`?

File: scripts/generate-config-data.js [hunk 3]
```diff
@@ -245,12 +255,12 @@ function augment (data, extra) {
...
         })
       })
-  );
+  ).filter(({props}) => props.length);
 
   // Hydrate config structs with CLI options
 
   const data = hydrateConfigWithCli(configSections, configCliOptions);
-
+
```
Comment (dvdplm) [inline scripts/generate-config-data.js]: extra whitespace

**Review by dvdplm — APPROVED** [review]
lgtm modulo minor nits

**Review by tomusdrw — APPROVED** [review]
lgtm. Merge at will after addressing @dvdplm grumbles.

File: scripts/generate-config-data.js (hunk 2 above)
Comment (axelchalon) [reply scripts/generate-config-data.js:130]: needs to match for example

```
	#[serde(rename="public_node")]
```

so we first need to trim() to remove the indentation and then check the starting chracters

**Pull request merged**